src/components/settings/clusters/rancher/Rancher.tsx

```diff
@@ -0,0 +1,230 @@
+
+  const handleApiTokenGeneration = async () => {
+    try {
+      if (username == '' || password == '') {
+        setError('Please enter Username and Password first!');
+        return;
+      }
+
+      const tokenResponse: IRancherTokenResponse = await getRancherToken(
+        rancherHost,
+        rancherPort,
+        username,
+        password,
+        bearerToken,
+        secure,
+      );
+
+      storeRancherCredentials();
+
+      setBearerToken(tokenResponse.token);
+      setUsername('');
+      setPassword('');
+    } catch (err) {
+      setError(err);
+    }
+  };
+
+  return (
+    <IonCard>
+      <div className="card-header-image">
+        <img alt="Rancher" src="/assets/card-header-rancher.png" />
+      </div>
+      <IonCardHeader>
+        <IonCardTitle>Import from Rancher Platform</IonCardTitle>
+      </IonCardHeader>
+
+      <IonCardContent>
+        <p className="paragraph-margin-bottom">
+          Choose this option to import your Rancher clusters from the Rancher Platform.
+        </p>
+
+        <IonList className="paragraph-margin-bottom" lines="full">
+          <IonItem>
+            <IonLabel position="stacked">Rancher Host</IonLabel>
+            <IonInput type="text" required={true} value={rancherHost} onInput={handleRancherHost} />
+          </IonItem>
+          <IonItem>
+            <IonLabel position="stacked">Rancher Port</IonLabel>
+            <IonInput type="number" required={true} value={rancherPort} onInput={handleRancherPort} />
+          </IonItem>
+          <IonItem>
+            <IonLabel>Secure</IonLabel>
+            <IonToggle checked={secure} onIonChange={handleSecure} />
+          </IonItem>
+          <IonItem>
+            <IonLabel position="stacked">Username</IonLabel>
+            <IonInput autocomplete="username" type="text" required={true} value={username} onInput={handleUsername} />
+          </IonItem>
+          <IonItem>
+            <IonLabel position="stacked">Password</IonLabel>
+            <IonInput
+              autocomplete="current-password"
+              type="password"
+              required={true}
+              value={password}
+              onInput={handlePassword}
+            />
+          </IonItem>
+          <IonItem>
+            <IonLabel position="stacked">Bearer Token (optional)</IonLabel>
+            <IonInput type="text" required={false} value={bearerToken} onInput={handleBearerToken} />
+            <IonButton
+              hidden={bearerToken != ''}
+              expand="full"
+              slot="end"
+              fill="clear"
+              shape="round"
+              size="large"
+              style={
+                isPlatform('ios')
+                  ? { '--padding-end': '0em' }
+                  : { '--padding-start': '0.5em', '--padding-end': '0.5em' }
+              }
+              onClick={() => setShowActionSheet(true)}
+            >
+              <IonIcon size="large" src="/assets/icons/misc/add-outline.svg" />
```

Comment:
Can we use an icon from ionicons here, maybe sth. like `import { add } from 'ionicons/icons';`

```suggestion
              <IonIcon icon={add} />
```

Reply:
Suggested change implemented.